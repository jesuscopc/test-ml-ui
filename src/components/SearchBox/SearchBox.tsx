import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import icSearch from 'assets/images/ic_Search.png';
import './index.scss';

const SearchBox = (): React.ReactElement => {
  const [inputSearch, setInputSearch] = useState<string>('');
  const history = useHistory();

  const handleSubmit = (e: React.FormEvent): void => {
    e.preventDefault();
    history.push(`/items?search=${inputSearch}`);
  };

  return (
    <form className="searchBox" onSubmit={handleSubmit}>
      <input
        type="search"
        placeholder="Nunca dejes de buscar"
        name="search"
        value={inputSearch}
        onChange={e => setInputSearch(e.currentTarget.value)}
      />
      <button type="submit">
        <img src={icSearch} alt="buscar" />
      </button>
    </form>
  );
};

export default SearchBox;
