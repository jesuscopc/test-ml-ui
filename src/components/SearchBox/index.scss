.searchBox {
  display: flex;
  & input {
    padding: 5px;
    width: 100%;
    font-size: 18px;
    border: none;
  }
  & button {
    border: none;
    background-color: #eeeeee;
    width: 50px;
  }
}
