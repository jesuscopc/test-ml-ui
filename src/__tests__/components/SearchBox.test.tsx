/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from '../../components/SearchBox/SearchBox';

describe('Suit Test SearchBox Component', () => {
  const handleSubmit = jest.fn();
  let wrapper = shallow(<SearchBox />);
  beforeEach(() => {
    wrapper = shallow(<SearchBox />);
  });

  test('should be render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be change inputTextValue', () => {
    const input = wrapper.find('input');
    const value = 'CHange Text';
    input.simulate('change', { currentTarget: { value } });
    expect(wrapper.find('input').props().value).toBe(value);
  });

  test('should send fomaValue form', () => {
    const history = jest.fn();
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(handleSubmit).not.toHaveBeenCalled();
  });
});
