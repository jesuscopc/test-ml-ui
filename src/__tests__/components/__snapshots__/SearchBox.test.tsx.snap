// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suit Test SearchBox Component should be render correctly 1`] = `
<form
  className="searchBox"
  onSubmit={[Function]}
>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Nunca dejes de buscar"
    type="search"
    value=""
  />
  <button
    type="submit"
  >
    <img
      alt="buscar"
      src={Object {}}
    />
  </button>
</form>
`;
